### PR TITLE
Rename interpreter to datastore.

### LINKED ISSUE #29: Rename interpreter
The fact that davenport is implemented using Scala free monads seems like a detail that could be hidden. The name "interpreter" doesn't seem to help much with comprehending the package and how to use it.

### PR DESCRIPTION
**Pull request by coltfred**
Fixes #29. Review to ensure I didn't miss anything please.

### CODE REVIEW

**Comment by zmre** [comment]
Looks fine to me.  I didn't spot any issues or missed spots.  Will wait for @BobWall23 to weigh in.

**Comment by BobWall23** [comment]
Looks good to me.

File: src/main/scala/com/ironcorelabs/davenport/datastore/MemDatastore.scala [hunk 1]
```diff
@@ -10,15 +10,15 @@ import scalaz._
 import scalaz.syntax.either._
 import DB._
 
-abstract class MemInterpreter extends Interpreter {
-  import MemInterpreter._
+abstract class MemDatastore extends Datastore {
+  import MemDatastore._
   protected var map: KVMap
```
Comment (zmre) [inline src/main/scala/com/ironcorelabs/davenport/datastore/MemDatastore.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/IronCore/davenport/file/1334834118/issues/source?bid=2825752&fileBranchId=2958426#l15)

Comment (zmre) [inline src/main/scala/com/ironcorelabs/davenport/datastore/MemDatastore.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/IronCore/davenport/file/1334834118/issues/source?bid=2825752&fileBranchId=2958426#l15)

File: src/main/scala/com/ironcorelabs/davenport/datastore/MemDatastore.scala [hunk 2]
```diff
@@ -29,18 +29,18 @@ abstract class MemInterpreter extends Interpreter {
...
 }
 
-object MemInterpreter {
+object MemDatastore {
   /** Backend of the memory store is a Map from Key -> RawJsonString */
   type KVMap = Map[Key, RawJsonString]
   type KVState[A] = StateT[Task, KVMap, A]
 
-  def apply(m: KVMap): MemInterpreter = new MemInterpreter {
+  def apply(m: KVMap): MemDatastore = new MemDatastore {
```
Comment (zmre) [inline src/main/scala/com/ironcorelabs/davenport/datastore/MemDatastore.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/IronCore/davenport/file/1334834118/issues/source?bid=2825752&fileBranchId=2958426#l38)

Comment (zmre) [inline src/main/scala/com/ironcorelabs/davenport/datastore/MemDatastore.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/IronCore/davenport/file/1334834118/issues/source?bid=2825752&fileBranchId=2958426#l38)

**Pull request merged**